modular_zubbers/code/modules/credits/credits.dm

```diff
@@ -0,0 +1,193 @@
+#define CREDIT_ROLL_SPEED 100
+#define CREDIT_SPAWN_SPEED 20
+#define CREDIT_ANIMATE_HEIGHT (14 * world.icon_size)
+#define CREDIT_EASE_DURATION 22
+GLOBAL_VAR_INIT(end_credits_song, null)
+GLOBAL_VAR_INIT(end_credits_title, null)
+GLOBAL_LIST(end_titles)
+#define JOINTEXT(X) jointext(X, null)
+/datum/controller/subsystem/ticker/declare_completion()
+	for(var/client/C)
+		if(!C.credits)
+			C.RollCredits()
+	for(var/thing in GLOB.clients)
+		var/client/C = thing
+		if (!C)
+			continue
+	. = ..()
+
+/datum/controller/subsystem/ticker/show_roundend_report(client/C, report_type = null)
+	. = ..()
+	for(var/iteration in GLOB.clients)
+		var/client/clients = iteration
+		if(!clients.credits)
+			clients.RollCredits()
+/* /client
+	var/list/credits */
+
+
+/client/proc/RollCredits()
+	set waitfor = FALSE
+
+
+	if(!GLOB.end_titles)
+		GLOB.end_titles = generate_titles()
```

Comment:
These should be generated before any calls to RollCredits() unless there's a good reason not to.